app/RootsTableViewController.m

```diff
@@ -90,7 +90,11 @@ - (void)documentPicker:(UIDocumentPickerViewController *)controller didPickDocum
 
     dispatch_async(dispatch_get_global_queue(QOS_CLASS_USER_INITIATED, 0), ^{
         NSError *error;
-        BOOL success = [Roots.instance importRootFromArchive:urls[0] name:name error:&error progressReporter:progressVC];
+        BOOL success = NO;
+        if ([url startAccessingSecurityScopedResource]) {
```

Comment:
error is used uninitialized if this is false

Reply:
It would be nil which should be fine in this case?

Reply:
it wouldn't, but realized that success is false so this is fine. Still, should we show something to the user if this fails? Not sure what failure here actually means though

Reply:
Oh I just spotted the `error != nil` check, we can delete that

Reply:
Not sure if importRootFromArchive can maybe return false with error null?

Reply:
In any case when something fails we should tell the user that the import didn't succeed, even if we don't have any additional error information

Reply:
Makes sense, but we should do that without possibly reading uninitialized data

Reply:
I'm not seeing the uninitialized read

Reply:
startAccessingSecurityScopedResource returns no, then success is no, but error was never set and will be read in the if below, and then maybe by presentError

Reply:
Right, it will be read as nil which is fine here

Reply:
Today I learned objective-c auto initializes object pointer variables to nil https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/WorkingwithObjects/WorkingwithObjects.html#//apple_ref/doc/uid/TP40011210-CH4-SW22